Add new category form improvements

afercia opened the pull request:
This PR tries to improve a bit the accessibility of the mini-form used to add new categories. Further improvements should be addressed in separate issues, see #2582 and #2581.

- adds visible label elements to the input field and select
- add IDs using `withInstanceId `
- removes the input placeholder
- removes the "+" plus sign from the toggle: we don't want screen readers announce "plus" and it's inconsistent with other `button-link`s used in the sidebar, e.g. "set featured image"
- fixes the focus styles 
- adds missing `type="text"` to the input field (this also fixes the focus style)
- keeps the form open after a category has been added: otherwise, when adding multiple categories, users would be forced to re-open the form each time
- adds `aria-expanded` to the toggle
- tries to improve translations and labels

Fixes #2572

afercia committed:
Add missing labels and instanceId.

afercia committed:
Merge branch 'master' into update/add-new-category-form-improvements

afercia committed:
Keep the form open.

afercia committed:
Fix focus styles using proper CSS classes.

afercia committed:
Improve spacing and translations, remove the plus sign.

afercia committed:
Merge branch 'master' into update/add-new-category-form-improvements

youknowriad commented on editor/sidebar/post-taxonomies/hierarchical-term-selector.js:
```diff
@@ -114,7 +115,7 @@ class HierarchicalTermSelector extends Component {
 					adding: false,
 					formName: '',
 					formParent: '',
-					showForm: false,
```
I guess we could drop it entirely here

afercia commented:
Visual comparison of the form, from left to right: before, after, after with focused elements:

<img width="943" alt="screen shot 2017-08-29 at 15 29 46" src="https://user-images.githubusercontent.com/1682452/29825388-87f851de-8cd4-11e7-8133-d52a6289d5e9.png">

aduth committed:
Sidebar: Remove form state toggle after term save

Previously closed form after add, now leaves _as is_, so no need to explicitly assigned

aduth reviewed (approved):
I made suggested changes per https://github.com/WordPress/gutenberg/pull/2588/files#r135794709 in f769c91937e636742343a22408795f198b2af86c. Looks good 👍

afercia commented:
Thanks! Will split a few pending improvements in a new issue.
